service/matching/task_queue_partition_manager.go

```diff
@@ -855,16 +855,18 @@ func (pm *taskQueuePartitionManagerImpl) getPhysicalQueuesForAdd(
 		return nil, syncMatchQueue, nil, err
 	}
 
-	if directive.GetBuildId() == nil {
-		// This means the tasks is a middle task belonging to an unversioned execution. Keep using unversioned.
-		return pm.defaultQueue, pm.defaultQueue, nil, nil
-	}
-
 	userData, userDataChanged, err := pm.userDataManager.GetUserData()
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
+	if directive.GetBuildId() == nil {
+		// This means the tasks is a middle task belonging to an unversioned execution. Keep using unversioned.
```

Comment:
what is a "middle task"? might be worth defining the term in parentheses right after it

Reply:
the first line is an old comment, updated it.